src/Views/OperatingSystemView.vala

```diff
@@ -39,39 +39,52 @@ public class About.OperatingSystemView : Gtk.Grid {
             logo_icon_name = "distributor-logo";
         }
 
-#if WALLPAPER
-        var logo = new Hdy.Avatar (128, "", false) {
-            // In case the wallpaper can't be loaded, we don't want an icon or text
-            icon_name = "invalid-icon-name",
-            // We need this for the shadow to not get clipped by Gtk.Overlay
-            margin = 6
-        };
-        logo.set_image_load_func ((size) => {
-            try {
-                return new Gdk.Pixbuf.from_file_at_scale ("/usr/share/backgrounds/elementaryos-default", -1, size, true);
-            } catch (Error e) {
-                critical (e.message);
-            }
-        });
-        logo.get_style_context ().add_provider (style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
-#endif
-
         var icon = new Gtk.Image () {
-            icon_name = logo_icon_name + "-symbolic",
-            // 128 minus 3px padding on each side
-            pixel_size = 128 - 6
+            icon_name = logo_icon_name,
         };
 
-        unowned var icon_style_context = icon.get_style_context ();
-        icon_style_context.add_class ("logo");
-        icon_style_context.add_provider (style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        var logo_overlay = new Gtk.Overlay ();
 
 #if WALLPAPER
-        var logo_overlay = new Gtk.Overlay ();
-        logo_overlay.add (logo);
-        logo_overlay.add_overlay (icon);
+        if (Gtk.IconTheme.get_default ().has_icon (logo_icon_name + "-symbolic")) {
+            foreach (unowned var path in Environment.get_system_data_dirs ()) {
+                var file = File.new_for_path (
+                    Path.build_path (Path.DIR_SEPARATOR_S, path, "backgrounds", "elementaryos-default")
+                );
+
+                if (file.query_exists ()) {
+                    var file_icon = new FileIcon (file);
+
+                    var logo = new Hdy.Avatar (128, "", false) {
+                        // In case the wallpaper can't be loaded, we don't want an icon or text
+                        icon_name = "invalid-icon-name",
```

Comment:
Do we need this line?  It just seems to produce lots of terminal critical warnings as "invalid-icon-name" cannot be found.  Leaving this line out had no noticeable effect for me.

Reply:
Yeah that should be okay to remove since we're checking for the existence of the file before constructing this